src/cpp/session/modules/SessionSource.cpp

```diff
@@ -1192,11 +1192,17 @@ SEXP rs_fileEdit(SEXP fileSEXP)
       if (!r::sexp::isString(fileSEXP))
          throw r::exec::RErrorException("invalid filename specification");
 
-      // extract string vector
+      // extract string vector (be careful to handle encodings)
       std::vector<std::string> filenames;
-      Error error = r::sexp::extract(fileSEXP, &filenames);
-      if (error)
-         throw r::exec::RErrorException(error.summary());
+      for (int i = 0; i < r::sexp::length(fileSEXP); i++)
+      {
+         SEXP filenameSEXP = STRING_ELT(fileSEXP, i);
+         cetype_t encoding = r::sexp::getEncoding(filenameSEXP);
+         if (encoding == CE_UTF8)
+            filenames.push_back(CHAR(filenameSEXP));
+         else
+            filenames.push_back(string_utils::systemToUtf8(CHAR(filenameSEXP)));
```

Comment:
Is there a reason we prefer the system encoding here? (seems like we might want to invoke r::sexp::extract and pass `utf8`?)

Reply:
Good point. I decided to just make that change (wire `asUtf8` through all of the string extraction APIs).